src/services/api/index.ts

```diff
@@ -0,0 +1,29 @@
+import axios from "axios";
+import { getSession } from "next-auth/react";
+
+const axiosInstance = axios.create({
+    baseURL: "http://localhost:3001/api/v1",
+    headers: { "Content-Type": "application/json" },
+});
+
+axiosInstance.interceptors.request.use(async (request) => {
+    const session = await getSession();
+
+    if (session) {
+        request.headers.Authorization = `Bearer ${session.accessToken}`;
+    }
+
+    return request;
+});
+
+export const api = axiosInstance;
+
+export const errorCatch = (error: any): string => {
+    const message = error?.response?.data?.message;
+
+    return message
+        ? typeof error.response.data.message === "object"
+            ? message[0]
+            : message
+        : error.message;
+};
```

Comment:
I think this is supposed to get message from errors.
There is a utility function `isAxiosError` in `axios` library, you can use it.

I usually create some file `services/error/helpers.ts`
where I add something like this:

```ts
import { isAxiosError } from 'axios'

export const ErrorHelpers = {
  getMessage: (error: unknown) => {
    if (isAxiosError<{ message: string | string[] }>(error)) {
      if (error.response?.data?.message) {
        const { message } = error.response.data
        return Array.isArray(message) ? message.join(', ') : message
      }
      return undefined
    }
    if (error instanceof Error) {
      return error.message
    }
    return undefined
  },
}

```